x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/TransportSqlClearCursorAction.java

```diff
@@ -47,7 +47,7 @@ public static void operation(
         SqlClearCursorRequest request,
         ActionListener<SqlClearCursorResponse> listener
     ) {
-        Cursor cursor = Cursors.decodeFromStringWithZone(request.getCursor()).v1();
+        Cursor cursor = Cursors.decodeFromStringWithZone(request.getCursor(), planExecutor.writeableRegistry()).v1();
```

Comment:
Is there a case where the Cursors would not all use the same internal writeableRegistry regardless of the user request?
That is, why isn't it a constant and has to be passed in as a parameter?

Reply:
It's mostly because of `SearchSourceBuilder` (in `CompositeAggCursor`)